### ISSUE TITLE
fix commits are based on `HEAD` of target branch, not PR

### ISSUE THREAD

**Issue by scarf005** [autofix-ci/action#12]
## Summary

```
- A (merge base) - B (remote HEAD)
   \
	C (PR HEAD) - C' (auto-fixed commit)
```

when `autofix.ci` creates commit `C'` from `C`, it seems to apply changes on top of `B`, not `C`.

## Example

let's assume repo for above graph 
- only tracks `file.txt`
- uses formatter that converts lines to `CapitalCase`, so `foo` -> `Foo`

on commit `A`:
```diff
1. Hello
2. World
```

on commit `B`:
```diff
- 1. Hello
2. World
```

on commit `C`:

```diff
1. Hello
2. World
+ 3. person
```

on commit `C'`, the auto-fixed one also applies changes from commit `B`:

```diff
- 1. Hello
2. World
- 3. person
+ 3. Person
```

I find this a bit odd, because i was expecting the formatting to be localized to `C`, like:

```diff
1. Hello
2. World
- 3. person
+ 3. Person
```

### [before formatting](https://github.com/cataclysmbnteam/Cataclysm-BN/commit/1ba67e2c9597288575a9c50a7b0facda91ed6841)

![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/54838975/408ca62b-d3b6-4437-af1b-7ec9ed502711)

In `locations_commercial.json`, line 343 is inserted.

### [During formatting](https://github.com/cataclysmbnteam/Cataclysm-BN/actions/runs/7326934945/job/19953170427?pr=3981)

```diff
diff --git a/data/json/itemgroups/Locations_MapExtras/locations_commercial.json b/data/json/itemgroups/Locations_MapExtras/locations_commercial.json
index faa4415..17dcf92 100644
--- a/data/json/itemgroups/Locations_MapExtras/locations_commercial.json
+++ b/data/json/itemgroups/Locations_MapExtras/locations_commercial.json
@@ -337,7 +337,7 @@
       [ "bandolier_shotgun", 8 ],
       [ "torso_bandolier_shotgun", 3 ],
       [ "holster", 8 ],
-	  [ "shoulder_holster", 4 ],
+      [ "shoulder_holster", 4 ],
       [ "bootstrap", 2 ],
       [ "wristholster", 1 ],
       [ "bholster", 5 ],
```

[In the log](https://github.com/cataclysmbnteam/Cataclysm-BN/actions/runs/7326934945/job/19953170427?pr=3981#step:9:62), we can find out that for `locations_commercial.json`, only line 343 is formatted.

## [After formatting](https://github.com/cataclysmbnteam/Cataclysm-BN/commit/40fcd788735a254c1992cfa1718d764b5b676ad9)

![image](https://github.com/autofix-ci/action/assets/54838975/60c2933b-b789-4134-8740-32483289519b)

However, [in the commit `autofix.ci`](https://github.com/cataclysmbnteam/Cataclysm-BN/blob/1ba67e2c9597288575a9c50a7b0facda91ed6841/data/json/itemgroups/Locations_MapExtras/locations_commercial.json#L14-L16) has made, line 14 to 16 is also removed. 

![image](https://github.com/autofix-ci/action/assets/54838975/23b4c74e-71d9-415b-a793-7ce9a7284938)

<details>
<summary>corresponding commit URLs since the PR branch is now deleted</summary>

- **merge-base** [fix(content): Update ergonomic grips to work for shotguns (#3974)](https://github.com/cataclysmbnteam/Cataclysm-BN/commit/b678b3674715562f63330078c9386ab25e4fbb39)
- **repo-head-back-then** [fix(content): Remove Rivtech guns from spawning in various commercial locations (#3975)](https://github.com/cataclysmbnteam/Cataclysm-BN/commit/6a6f2901f499b230615f3d2e3880a2e9305ff7ae)
- [feat(content): shoulder holsters](https://github.com/Relgar/Cataclysm-BN/commit/1ba67e2c9597288575a9c50a7b0facda91ed6841)
- **autofix-oddity** [style(autofix.ci): automated formatting](https://github.com/Relgar/Cataclysm-BN/commit/40fcd788735a254c1992cfa1718d764b5b676ad9)

</details>

This is unexpected because the [PR's merge base](https://github.com/cataclysmbnteam/Cataclysm-BN/commit/b678b3674715562f63330078c9386ab25e4fbb39) was older than [its removal](https://github.com/cataclysmbnteam/Cataclysm-BN/commit/6a6f2901f499b230615f3d2e3880a2e9305ff7ae).

## Additional Context

https://github.com/cataclysmbnteam/Cataclysm-BN/pull/3981#issuecomment-1869278777

**Comment by mhils** [comment]
Ughs, thanks for catching this! This definitely is a bug in the action right now. I guess we need to do something like this to fix it:

 1. Create commit with all changes.
 2. Check out PR head.
 3. Cherry-pick the commit from step 1.
 4. Generate autofix.json as we already do.

**Comment by mhils** [comment]
This has been fixed in [dd55f44](https://github.com/autofix-ci/action/commit/dd55f44df8f7cdb7a6bf74c78677eb8acd40cd0a) :)
